src/resmom/linux/alps.c

```diff
@@ -3267,42 +3281,66 @@ inventory_to_vnodes(basil_response_t *brp)
 				NULL) == -1)
 				goto bad_vnl;
 
-			attr = "resources_available.ncpus";
-			totcpus = 0;
-			for (proc=seg->processors; proc; proc=proc->next)
-				totcpus++;
-			sprintf(utilBuffer, "%d", totcpus);
-			if (vn_addvnr(nv, vname, attr, utilBuffer,
-				0, 0, NULL) == -1)
-				goto bad_vnl;
-
-			attr = "resources_available.mem";
-			totmem = 0;
-			for (mem=seg->memory; mem; mem=mem->next)
-				totmem += mem->page_size_kb * mem->page_count;
-			sprintf(utilBuffer, "%ldkb", totmem);
-			if (vn_addvnr(nv, vname, attr, utilBuffer,
-				0, 0, NULL) == -1)
-				goto bad_vnl;
+			if (vnode_per_numa_node) {
+				attr = "resources_available.ncpus";
+				totcpus = 0;
+				for (proc=seg->processors; proc; proc=proc->next)
+					totcpus++;
+				snprintf(utilBuffer, sizeof(utilBuffer), "%d",
+					totcpus);
+				if (vn_addvnr(nv, vname, attr, utilBuffer,
+					0, 0, NULL) == -1)
```

Comment:
Why do we pass attribute type and flags as 0 here and a few places below like in case of mem? Shouldn't the type be ATR_TYPE_LONG for ncpus and ATR_TYPE_SIZE for mem and flag as atype?